core/src/env.rs

```diff
@@ -368,20 +328,10 @@ mod tests {
     #[test]
     /// Test that ext's `with<R>(...)` works correct when the inner is set
     fn calling_fn_with_inner_ext() {
-        let mut ext: LaterExt<_> = Default::default();
-        ext.set(ExtImplementedStruct(0));
+        let ext = LaterExt::new(ExtImplementedStruct(0));
 
         let converted_inner = ext.with(converter);
 
         assert!(converted_inner.is_ok());
     }
```

Comment:
Where is the test we discussed about cloning and taking? There is a proc macro `#[should_panic]` for such cases

Reply:
Will be in the next PR. Currently we return only option, so no need for panic checks.